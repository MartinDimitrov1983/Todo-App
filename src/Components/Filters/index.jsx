import React from 'react';
import { CheckBox, ColorDot, Button } from '..';
import {
  CLEAR_FILTERS,
  FILTERS,
  MY_TASKS,
  RESOLVED,
  UNRESOLVED,
  COLORS,
} from '../../Constants';
import styles from './index.module.css';

const Filters = ({
  setResolved,
  setUnresolved,
  setFilterColor,
  resolved,
  unresolved,
  dotColors,
}) => {
  return (
    <>
      <h2 className={styles.title}>{MY_TASKS}</h2>
      <div>
        <div className={styles.filters}>
          <h4>{FILTERS}</h4>
          <Button
            onClick={() => {
              setResolved(false);
              setUnresolved(false);
              setFilterColor(false);
            }}
            text={CLEAR_FILTERS}
          />
        </div>
        <div className={styles.filter}>
          <CheckBox
            label={RESOLVED}
            checked={resolved}
            onChange={() => setResolved((prev) => !prev)}
          />
          <CheckBox
            label={UNRESOLVED}
            checked={unresolved}
            onChange={() => setUnresolved((prev) => !prev)}
          />
        </div>
        <div>
          <h5>{COLORS}</h5>
          <div className={styles.colorFilter}>
            {dotColors.map((color) => (
              <ColorDot
                color={color}
                key={color}
                onClick={() => setFilterColor(color)}
              />
            ))}
          </div>
        </div>
      </div>
    </>
  );
};

export default Filters;
