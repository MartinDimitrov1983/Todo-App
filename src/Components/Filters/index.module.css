.title {
  text-align: center;
}

.filter {
  display: flex;
  gap: 20px;
}

.colorFilter {
  display: flex;
  margin-bottom: 20px;
  gap: 10px;
}

.filters {
  display: flex;
  justify-content: space-between;
  align-items: center;
}
