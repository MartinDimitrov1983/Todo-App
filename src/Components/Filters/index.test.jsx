import React from 'react';
import { render } from '@testing-library/react';

import Filters from './index';

test('Snapshot Filters', () => {
  const { asFragment } = render(
    <Filters
      resolved={false}
      unresolved={false}
      dotColors={[]}
      setResolved={() => {}}
      setUnresolved={() => {}}
      setFilterColor={() => {}}
    />,
  );
  expect(asFragment()).toMatchSnapshot();
});
