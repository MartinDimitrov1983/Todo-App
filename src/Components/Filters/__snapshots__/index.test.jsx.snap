// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Filters 1`] = `
<DocumentFragment>
  <h2
    class="title"
  >
    My Tasks
  </h2>
  <div>
    <div
      class="filters"
    >
      <h4>
        Filters
      </h4>
      <button
        class="button"
        type="button"
      >
        <div>
          Clear Filters
        </div>
      </button>
    </div>
    <div
      class="filter"
    >
      <div
        class="checkBoxWrapper"
      >
        <label
          class="label"
        >
          <input
            class="input"
            type="checkbox"
          />
          <span>
            Resolved
          </span>
        </label>
      </div>
      <div
        class="checkBoxWrapper"
      >
        <label
          class="label"
        >
          <input
            class="input"
            type="checkbox"
          />
          <span>
            Unresolved
          </span>
        </label>
      </div>
    </div>
    <div>
      <h5>
        Colors
      </h5>
      <div
        class="colorFilter"
      />
    </div>
  </div>
</DocumentFragment>
`;
